torch/serialization.py

```diff
@@ -748,11 +748,11 @@ def persistent_load(saved_id):
         # only if offset is zero we can attempt the legacy tar file loader
         try:
             return legacy_load(f)
-        except tarfile.TarError:
+        except tarfile.TarError as e:
             if _is_zipfile(f):
                 # .zip is used for torch.jit.save and will throw an un-pickling error here
                 raise RuntimeError(
-                    "{filename} is a zip archive (did you mean to use torch.jit.load()?)".format(filename=f.name))
+                    "{filename} is a zip archive (did you mean to use torch.jit.load()?)".format(filename=f.name)) from e
```

Comment:
No need to chain the original error here (also use f-string for formatting)
```suggestion
                    f"{f.name} is a zip archive (did you mean to use torch.jit.load()?)") from None
```